src/main/groovy/actions/TransferActions.groovy

```diff
@@ -4,41 +4,102 @@
-        if (source.labwareType.layout != destination.labwareType.layout)
-            throw new TransferException("Labwares must have the same layout. ${source.labwareType.layout.name} and ${destination.labwareType.layout.name}")
+    def static stamp(Labware sourceLabware, Labware destinationLabware, 
+        MaterialType materialType, List<String> copyMetadata = []) {
 
-        def sourceMaterialUuidToLocation = source.receptacles.collectEntries { [it.materialUuid, it.location] }
+        if (sourceLabware.labwareType.layout != destinationLabware.labwareType.layout)
+            throw new TransferException("Labwares must have the same layout. ${sourceLabware.labwareType.layout.name} and ${destinationLabware.labwareType.layout.name}")
 
-        def sourceMaterials = MaterialActions.getMaterials(source.materialUuids())
+        def sourceMaterialUuidToLocation = sourceLabware.receptacles.collectEntries { [it.materialUuid, it.location] }
+
+        def sourceMaterials = getMaterialsFromLabware(sourceLabware.materialUuids())
         def destinationMaterials = sourceMaterials.collect { sourceMaterial ->
-            new Material(
-                name: "${destination.barcode}_${sourceMaterialUuidToLocation[sourceMaterial.id].name}",
-                materialType: materialType,
-                metadata: sourceMaterial.metadata.findAll { it.key in copyMetadata },
-                parents: [sourceMaterial]
-            )
+            createNewChildMaterial("${destinationLabware.barcode}_${sourceMaterialUuidToLocation[sourceMaterial.id].name}", 
+                materialType, sourceMaterial, copyMetadata)
         }
-        destinationMaterials = MaterialActions.postMaterials(destinationMaterials)
+        destinationMaterials = postNewMaterials(destinationMaterials)
 
         destinationMaterials.each { material ->
             def location = sourceMaterialUuidToLocation[material.parents[0].id]
-            def receptacle = destination.receptacles.find { it.location == location }
+            def receptacle = destinationLabware.receptacles.find { it.location == location }
             receptacle.materialUuid = material.id
         }
 
-        LabwareActions.updateLabware(destination)
+        updateLabware(destinationLabware)
+    }
+
+    def static split(Labware sourceLabware, Labware destinationLabware, MaterialType materialType,
+        List<String> destinationLocations, List<String> copyMetadata = []) {
+
+        def destinationLabwareLocations =
+            destinationLabware.receptacles.collect { it.location.name }
+        def missingLocations = new ArrayList<String>()
+        destinationLocations.each { destinationLocation ->
+            if (!destinationLabwareLocations.contains(destinationLocation)) {
+                missingLocations.add(destinationLocation)
+            }
+        }
+        if (missingLocations.size > 0) {
```

Comment:
Under the hood Groovy is using the size() method, but I can change it to use it directly